cmake/sycl/CMakeTestSYCLCompiler.cmake

```diff
@@ -17,7 +17,7 @@ endif()
 # Try to use the HIP compiler.
 file( WRITE
    "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/main.sycl"
-   "#include <CL/sycl.hpp>\n"
+   "#include <sycl/sycl.hpp>\n"
```

Comment:
This is the same across all implementations now?

Reply:
As I've been told, the [SYCL2020 standard](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html) specifies this as **the** SYCL include.

https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:headers-and-namespaces

At the same time the standard also doesn't explicitly deprecate `<CL/sycl.hpp>` with the corresponding namespace. 😕 But apparently Intel wants to do that.

Since implicitly we've been relying on SYCL2020 since forever, we're now just making it (more) explicit. 🤔